engine/src/main/java/io/camunda/zeebe/engine/state/instance/ElementInstance.java

```diff
@@ -22,6 +22,7 @@
 
   private final LongProperty parentKeyProp = new LongProperty("parentKey", -1L);
   private final IntegerProperty childCountProp = new IntegerProperty("childCount", 0);
+  private final IntegerProperty numberOfInstancesProp = new IntegerProperty("numberOfInstances", 0);
```

Comment:
🔧  I think `childActivatedCount` would be a better name in this context. Firstly, it could read as if the number represents all instances of this element instead of children of this specific element instance. And secondly, because it stands next to `childCount` and `childActivatedCount` would be somewhat consistent naming. If we expand it with `childCompletedCount` and `childTerminatedCount` in the future, it's somewhat easier to understand that `childCount` is the `childActiveCount`. 

To apply this further, we can already rename the field name from `childCountProp` to `childActiveCountProp`, but we cannot change the `childCount` name as that would be a breaking change.